src/components/Form.js

```diff
@@ -54,18 +54,43 @@ export default class Form extends Component {
     this.setState(nextState);
   }
 
-  getStateFromProps(props, inputFormData) {
+  getStateFromProps(nextProps, inputFormData) {
     const state = this.state || {};
-    const schema = "schema" in props ? props.schema : this.props.schema;
-    const uiSchema = "uiSchema" in props ? props.uiSchema : this.props.uiSchema;
+
+    if (
+      nextProps &&
+      inputFormData === state.formData &&
+      nextProps.schema === this.props.schema &&
+      nextProps.uiSchema === this.props.uiSchema &&
+      nextProps.liveValidate === this.props.liveValidate &&
+      nextProps.customFormats === this.props.customFormats &&
+      nextProps.additionalMetaSchemas === this.props.additionalMetaSchemas &&
+      nextProps.noValidate === this.props.noValidate &&
+      nextProps.idPrefix === this.props.idPrefix
+    ) {
+      // No data has changed, maintain state.
+      return state;
+    }
+
+    const schema = nextProps ? nextProps.schema : this.props.schema;
+    const uiSchema = nextProps ? nextProps.uiSchema : this.props.uiSchema;
+    const liveValidate = nextProps
+      ? nextProps.liveValidate
+      : this.props.liveValidate;
+    const customFormats = nextProps
```

Comment:
Nice catch with `customFormats` and `additionalMetaSchemas` -- it didn't fall back to the props value previously. Do you know what kinds of bugs could have been caused by that omission previously? We may want to add tests that replicate those bugs and ensure that they don't regress.

Reply:
Shouldn't be any bugs, the only time partial props were being passed is from tests.  React itself will always give you every single property on componentWillReceiveProps, so in practice this check should be completely unnecessary.